code/modules/acting/acting_items.dm

```diff
@@ -24,14 +24,19 @@
 	anchored = 1
 	density = 1
 
-/obj/machinery/acting/changer/attack_hand(var/mob/user as mob)
-	if(ishuman(user))
-		var/mob/living/carbon/human/H = user
-		H.change_appearance(APPEARANCE_ALL, H, TRUE, H.generate_valid_species(), null, default_state, src)
-		var/getName = sanitizeName(sanitize_readd_odd_symbols(sanitize(input(H, "Would you like to change your name to something else?", "Name change") as null|text, MAX_NAME_LEN)))
-		if(getName)
-			H.real_name = getName
-			H.name = getName
-			H.dna.real_name = getName
-			if(H.mind)
-				H.mind.name = H.name
+/obj/machinery/acting/changer/attack_hand(var/mob/living/carbon/human/H)
+	if(!istype(H))
+		return
+
+	H.change_appearance(APPEARANCE_ALL, H, TRUE, H.generate_valid_species(), null, default_state, src)
+	var/getName = sanitizeName(sanitize_readd_odd_symbols(sanitize(input(H, "Would you like to change your name to something else?", "Name change") as null|text, MAX_NAME_LEN)))
```

Comment:
Why do we have a name length limit here but not in the other places this PR touches? (One of which you removed the limit in)

Reply:
just forgot to take it out i think